yente/provider/elastic.py

```diff
@@ -229,15 +229,28 @@ async def search(
             )
             raise YenteIndexError(f"Could not search index: {ae}") from ae
 
-    async def bulk_index(self, entities: AsyncIterator[Dict[str, Any]]) -> None:
-        """Index a list of entities into the search index."""
+    async def bulk_index(
+        self, entities: AsyncIterator[Dict[str, Any]]
+    ) -> Tuple[int, int]:
+        """Index a list of entities into the search index.
+
+        Args:
+            entities: An async iterator of entities to index.
+
+        Returns:
+            A tuple of the number of entities indexed and the number of errors.
+        """
         try:
-            await async_bulk(
+            n, errors = await async_bulk(
                 self.client(),
                 entities,
                 chunk_size=1000,
-                yield_ok=False,
-                stats_only=True,
+                raise_on_error=False,
```

Comment:
Does this mean that when there is an error, the indexer doesn't abort and still streams in a full set of entities? I kind of want to think through whether that has a way of backfiring (not obvious, but it is a change).

Reply:
That's what it means, yes - I made the change because we have stats_only set to True which doesn't do anything if raise_on_error isn't False. 

The logs shared in Slack show that some users are having issues with single records failing, so currently that would mean that the entire update (full or delta, doesn't matter) will fail and they will be stuck on whatever version they were on before. 
This way an error gets logged and the update succeeds partially which seems to happen quite a bit.
I see two potential issues with this:
1. The onus is on us to discover the single document errors and fix them. Since we don't have access to client logs we'd need to discover it in the logs of the ServiceAPI. This will be the case no matter what approach we take, but failing with no updates is a lot louder than partially succeeding.
2. There is an error in a delta update that would have worked if run with a full re-index. In this case they will be missing out on that operation and the error will be propagated forward since they're on delta updates.

As an aside, maybe we should have a "turn it off and on again" schedule? Every X days we do a full reindex instead of a delta update.

Reply:
If a single record fails to index, the index job has failed and we shouldn't migrate to that version. Anything else makes this software a liability nightmare for us and our customers. Can we please make sure that principle is restored and that there's no chance ever that a non-complete index is rolled into? 

Regarding the "lets do a full reindex every Nth time" - I like that, I just don't know where we'd store it. Doing a full special index for it would be a bit much.